drivers/platform/aducm3029/rtc_extra.h

```diff
@@ -0,0 +1,75 @@
+ *  - Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *  - Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#ifndef RTC_EXTRA_H_
+#define RTC_EXTRA_H_
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include <drivers/rtc/adi_rtc.h>
+
+/******************************************************************************/
+/*************************** Types Declarations *******************************/
+/******************************************************************************/
+
+/**
+ * @struct aducm_rtc_dev
+ * @brief Structure holding ADuCM3029 specific RTC descriptor.
+ */
+struct aducm_rtc_dev {
+	/** Pointer to the DFP driver handler. */
+	void *instance;
+	/** Pointer to the memory used by the DFP driver. */
+	void *memory;
+	/** Input clock prescaler. */
+	uint8_t prescaler;
+};
+
+/**
+ * @struct aducm_rtc_init_param
+ * @brief Structure holding ADuCM3029 specific RTC initialization parameters.
+ */
+struct aducm_rtc_init_param {
+	/** Input clock prescaler. */
+	uint8_t prescaler;
```

Comment:
I think you can use an enum with the available frequencies and calculate the prescaler inside the driver.
Also specify somewhere that only for the RTC1 a frequency can be set and for RTC0 is 1Hz. 
Maybe move this to generic file as frequency

Reply:
Will do.